packages/experiments/src/a11y/focus/Focus.ts

```diff
@@ -0,0 +1,243 @@
+   *    1. <button>, <input>, <select> or <textarea> element that is not disabled
+   *    2. <a> element with non-empty href attribute
+   *    3. contenteditable attribute set to true
+   *
+   * Special case for compliance with office-ui-fabric-react: office-ui-fabric-react uses data-is-focusable attribute
+   * for all elements that could be focusable inside a FocusZone and uses tabindex=0 to set the focus currently
+   * focused element and all other elements that that are not currently focused have tabindex=-1. This can hide those
+   * actually focusable elements from a standard search (because their tabindex is -1), therefore the ignoreTabIndex
+   * input parameter is provided to make sure those elements are discoverable. In case you have office-ui-fabric-react
+   * controls in your application, you may want to use this parameter depending on your scenario. Otherwise, you can
+   * ignore the ignoreTabIndex parameter.
+   */
+  public static isElementFocusable(elem: HTMLElement, ignoreTabIndex: boolean = false): boolean {
+    if (
+      elem
+      && !elem.hidden
+      && elem.offsetParent
+      && window.getComputedStyle(elem).display !== 'none'
+      && window.getComputedStyle(elem).visibility !== 'hidden'
+    ) {
+      const tagName: string = elem.tagName.toLowerCase();
+      const hasTabIndex: boolean = elem.hasAttribute('tabindex');
+      const tabindex: number = parseInt(elem.getAttribute('tabindex') || '', 10);
+
+      if (!ignoreTabIndex && hasTabIndex && tabindex < 0) {
+        return false;
+      }
+
+      if (
+        tagName === 'a' && !!(elem as HTMLAnchorElement).href ||
+        elem.isContentEditable
+      ) {
+        return true;
+      }
+
+      if (
+        (
+          tagName === 'button' ||
+          tagName === 'input' ||
+          tagName === 'select' ||
+          tagName === 'textarea'
+        ) && !(elem as HTMLButtonElement).disabled
+      ) {
+        return true;
+      }
+
+      /**
+       * There are few cases like office-ui-fabric-react's dropdown control, which do not
+       * use the conventional HTML elements to create dropdown. Instead they are using a
+       * data attribute 'is-focusable' and making it focusable. Hence to respect those elements
+       * added this check. Check this only when the 'ignoreTabIndex' is true, this is because when
+       * user doesn't want to ignore tabindexes then tabindex will get preference over this attribute.
+       */
+      if (ignoreTabIndex && elem.getAttribute('data-is-focusable') === 'true') {
+        return true;
+      }
+
+      if (!ignoreTabIndex && hasTabIndex && tabindex > -1) {
+        return true;
+      }
+    }
+
+    return false;
+  }
+
+  /**
+   * Gets all the immediate children of the given element's Focusable Sub-Tree. The order is the same as
+   * the default tab order of the elements.
+   */
+  public static getFocusableChildren(element: HTMLElement, ignoreTabIndex: boolean = false): HTMLElement[] {
```

Comment:
We don't use static methods. we just export functions. This helps with treeshaking.